src/main/scala/org/neo4j/spark/service/Neo4jQueryService.scala

```diff
@@ -1,39 +1,77 @@
        |""".stripMargin
 
-  override def createStatementForRelationships(options: Neo4jOptions): String = throw new UnsupportedOperationException("TODO implement method")
+  private def keywordFromSaveMode(saveMode: Any): String = {
+    saveMode match {
+      case NodeSaveMode.Overwrite | SaveMode.Overwrite => "MERGE"
+      case NodeSaveMode.ErrorIfExists | SaveMode.ErrorIfExists => "CREATE"
+      case NodeSaveMode.Match => "MATCH"
+      case _ => throw new UnsupportedOperationException(s"Source SaveMode $saveMode not supported")
+    }
+  }
+
+  override def createStatementForRelationships(options: Neo4jOptions): String = {
+    val relationshipKeyword = keywordFromSaveMode(saveMode)
+    val sourceKeyword = keywordFromSaveMode(options.relationshipMetadata.sourceSaveMode)
+    val targetKeyword = keywordFromSaveMode(options.relationshipMetadata.targetSaveMode)
+
+    val relationship = options.relationshipMetadata.relationshipType.quote()
+
+    val sourceLabels = options.relationshipMetadata.source.labels
+      .map(_.quote())
+      .mkString(":")
+
+    val targetLabels = options.relationshipMetadata.target.labels
+      .map(_.quote())
+      .mkString(":")
+
+    val sourceKeys = options.relationshipMetadata.source.nodeKeys.map(key => {
+      s"${key._2.quote()}:$BATCH_VARIABLE.source.${Neo4jWriteMappingStrategy.KEYS}.${key._1.removeAlias().quote()}"
+    }).mkString(", ")
+    val targetKeys = options.relationshipMetadata.target.nodeKeys.map(key => {
+      s"${key._2.quote()}:$BATCH_VARIABLE.target.${Neo4jWriteMappingStrategy.KEYS}.${key._1.removeAlias().quote()}"
+    }).mkString(", ")
+
+    val sourceQueryPart = createQueryPart(sourceKeyword, sourceLabels, sourceKeys, Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS)
+    val targetQueryPart = createQueryPart(targetKeyword, targetLabels, targetKeys, Neo4jUtil.RELATIONSHIP_TARGET_ALIAS, Seq(Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS))
+
+    s"""UNWIND ${"$"}events AS $BATCH_VARIABLE
+       |$sourceQueryPart
+       |$targetQueryPart
+       |$relationshipKeyword (${Neo4jUtil.RELATIONSHIP_SOURCE_ALIAS})-[${Neo4jUtil.RELATIONSHIP_ALIAS}:${relationship}]->(${Neo4jUtil.RELATIONSHIP_TARGET_ALIAS})
+       |SET ${Neo4jUtil.RELATIONSHIP_ALIAS} += $BATCH_VARIABLE.${Neo4jUtil.RELATIONSHIP_ALIAS}.${Neo4jWriteMappingStrategy.PROPERTIES}
+       |""".stripMargin
+  }
 
   override def createStatementForNodes(options: Neo4jOptions): String = {
-    val keyword = saveMode match {
-      case SaveMode.Overwrite => "MERGE"
-      case SaveMode.ErrorIfExists => "CREATE"
-      case _ => throw new UnsupportedOperationException(s"SaveMode $saveMode not supported")
-    }
+    val keyword = keywordFromSaveMode(saveMode)
+
     val labels = options.nodeMetadata.labels
       .map(_.quote)
       .mkString(":")
     val keys = options.nodeMetadata.nodeKeys.keys
       .map(_.quote)
-      .map(k => s"$k: event.keys.$k")
+      .map(k => s"$k: $BATCH_VARIABLE.${Neo4jWriteMappingStrategy.KEYS}.$k")
```

Comment:
needs backticks to account for spaces and special characters in property names too.   This needs to be abstracted out into a small function, "turn a string into a valid property name" or somesuch.

Reply:
the line `68` (`.map(_.quote)`) adds backticks so `k` is already "backticked" (=quoted)